sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -118,7 +118,66 @@ import org.apache.spark.sql.types.IntegerType
  *       LocalTableScan [...]
  * }}}
  *
- * The rule does the following things here:
+ * Third example: single distinct aggregate function with filter clauses (in sql):
```

Comment:
Can you replace this example with more general one. e.g, `COUNT(DISTINCT cat1) FILTER (WHERE id > 1), COUNT(DISTINCT cat2)`?

Reply:
OK